src/Descriptor.ts

```diff
@@ -0,0 +1,62 @@
+enum DescriptorKind {
+  Package,
+  Type,
+  Term,
+  Meta,
+  Method,
+  Parameter,
+  TypeParameter,
+}
+
+export class Descriptor {
+  private constructor(
+    public readonly name: string,
+    public readonly kind: DescriptorKind,
+    public readonly disambiguator?: string
+  ) {}
+  public static package(name: string): Descriptor {
+    return new Descriptor(name, DescriptorKind.Package)
+  }
+  public static type(name: string): Descriptor {
+    return new Descriptor(name, DescriptorKind.Type)
+  }
+  public static term(name: string): Descriptor {
+    return new Descriptor(name, DescriptorKind.Term)
+  }
+  public static meta(name: string): Descriptor {
+    return new Descriptor(name, DescriptorKind.Meta)
+  }
+  public static method(name: string, disambiguator: string): Descriptor {
```

Comment:
Maybe remove the `disambiguator` for now? I'm assuming it's there for method overloading -- we can re-add it later when we add support for method overloading. Right now, it seems a little strange because both call-sites are passing in `''`.